src/client/components/pages/Courses/CourseInstanceTableBody.tsx

```diff
@@ -0,0 +1,99 @@
+import { CellLayout } from 'client/components/general';
+import { COURSE_TABLE_COLUMN, getAreaColor, TERM } from 'common/constants';
+import CourseInstanceResponseDTO from 'common/dto/courses/CourseInstanceResponse';
+import {
+  TableBody, TableRow, TableRowHeadingCell, VALIGN, TableCell,
+} from 'mark-one';
+import React, { FunctionComponent, ReactElement, Ref } from 'react';
+import { CourseInstanceListColumn } from './tableFields';
+
+interface CourseInstanceTableBodyProps {
+  /**
+   * The list of courses to be shown in the table
+   */
+  courseList: CourseInstanceResponseDTO[];
+  /**
+   * The data to display
+   */
+  tableData: CourseInstanceListColumn[];
+  /**
+   * Controls the opening of the meeting modal with the requested course and term
+   */
+  openMeetingModal: (course: CourseInstanceResponseDTO, term: TERM) => void;
+  /**
+   * Controls the opening of the instructor modal with the requested course and term
+   */
+  openInstructorModal: (course: CourseInstanceResponseDTO, term: TERM) => void;
+  /**
+   * The ref value of the edit faculty absence button
+   */
+  buttonRef: Ref<HTMLButtonElement>;
```

Comment:
Might be worth renaming this param to `editButtonRef` or similar to make clear that it's for the edit faculty button.

Reply:
Jonathan ended up changing the way that refs are handled in this commit: [445e084](https://github.com/seas-computing/course-planner/pull/454/commits/445e084764a5aacd4e541f02cc52280cd8d9a691) and [5e6122d](https://github.com/seas-computing/course-planner/pull/454/commits/5e6122d6cdc734a3a2c8a46efa706ec1b6743b06).